integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBConfigNodeIT.java

```diff
@@ -0,0 +1,160 @@
+        ConfigFactory.getConfig().getSchemaRegionConsensusProtocolClass();
+    enableDataRegionConsensusProtocolClass =
+        ConfigFactory.getConfig().getDataRegionConsensusProtocolClass();
+
+    ConfigFactory.getConfig()
+        .setConfigNodeConsesusProtocolClass("org.apache.iotdb.consensus.ratis.RatisConsensus");
+    ConfigFactory.getConfig()
+        .setSchemaRegionConsensusProtocolClass("org.apache.iotdb.consensus.ratis.RatisConsensus");
+    ConfigFactory.getConfig()
+        .setDataRegionConsensusProtocolClass("org.apache.iotdb.consensus.ratis.RatisConsensus");
+
+    EnvFactory.getEnv().initBeforeClass();
+  }
+
+  @After
+  public void tearDown() {
+    EnvFactory.getEnv().cleanAfterClass();
+    ConfigFactory.getConfig()
+        .setConfigNodeConsesusProtocolClass(enableConfigNodeConsensusProtocolClass);
+    ConfigFactory.getConfig()
+        .setSchemaRegionConsensusProtocolClass(enableSchemaRegionConsensusProtocolClass);
+    ConfigFactory.getConfig()
+        .setDataRegionConsensusProtocolClass(enableDataRegionConsensusProtocolClass);
+  }
+
+  private TShowClusterResp getClusterNodeInfos(
+      IConfigNodeRPCService.Iface client, int configNodeNum, int dataNodeNum)
+      throws TException, InterruptedException {
+    TShowClusterResp clusterNodes = null;
+    for (int i = 0; i < retryNum; i++) {
+      clusterNodes = client.showCluster();
+      Thread.sleep(1000);
+
+      if (clusterNodes.getConfigNodeListSize() == configNodeNum
+          && clusterNodes.getDataNodeListSize() == dataNodeNum) break;
+    }
+
+    if (clusterNodes.getConfigNodeListSize() != configNodeNum
+        || clusterNodes.getDataNodeListSize() != dataNodeNum) {
+      fail("getClusterNodeInfos failed");
+    }
+
+    return clusterNodes;
+  }
+
+  @Test
+  public void removeAndStopConfigNodeTest() {
+    TShowClusterResp clusterNodes;
+    TSStatus status;
+
+    List<ConfigNodeWrapper> configNodeWrappers = EnvFactory.getEnv().getConfigNodeWrapperList();
+
+    try (SyncConfigNodeIServiceClient client =
+        (SyncConfigNodeIServiceClient) EnvFactory.getEnv().getConfigNodeConnection()) {
+      // add ConfigNode
+      for (int i = 0; i < 2; i++) {
+        ConfigNodeWrapper configNodeWrapper =
+            new ConfigNodeWrapper(
+                false,
+                configNodeWrappers.get(0).getIpAndPortString(),
+                "IoTDBConfigNodeIT",
+                "removeAndStopConfigNodeTest",
+                EnvUtils.searchAvailablePorts());
+        configNodeWrapper.createDir();
+        configNodeWrapper.changeConfig(ConfigFactory.getConfig().getConfignodeProperties());
+        configNodeWrapper.start();
+        configNodeWrappers.add(configNodeWrapper);
+      }
+      EnvFactory.getEnv().setConfigNodeWrapperList(configNodeWrappers);
+      clusterNodes = getClusterNodeInfos(client, 3, 3);
+      assertEquals(TSStatusCode.SUCCESS_STATUS.getStatusCode(), clusterNodes.getStatus().getCode());
```

Comment:
It's better to add more Asserts for `TShowClusterResp`. For example, assert NodeStatus, NodeId, NodeIp and NodePort